docs/CODE_GUIDELINES.md

```diff
@@ -688,15 +766,32 @@ For functions that have multiple output values, prefer using a `struct` or `tupl
 ### 12.2. Casts
 New code has to use C++ style casts and not older C style casts. When modifying existing code the developer can choose to update it to C++ style casts or leave as is. Whenever a `dynamic_cast` is used to cast to a pointer type, the result can be `nullptr` and needs to be checked accordingly.
 
+✅ Good:
+```cpp
+char m_dataChar{...};
+uint8_t m_dataInt = static_cast<uint8_t>(m_dataChar);
+```
+
+❌ Bad:
+```cpp
+char m_dataChar{...};
+uint8_t m_dataInt = (uint8_t)m_dataChar;
+```
+
 ### 12.3. `NULL` vs `nullptr`
 Prefer the use of `nullptr` instead of `NULL`. `nullptr` is a typesafe version and as such can't be implicitly converted to `int` or anything else.
 
 ### 12.4. `auto`
 
-Feel free to use `auto` wherever it improves readability, which is not always the case. Good places are iterators or when dealing with containers. Bad places are code that expects a certain type that is not immediately clear from the context.
+Feel free to use `auto` wherever it improves readability, without abusing it when it is not the case.
+- Good places are iterators or types that have multiple sub-levels in a namespace.
+- Bad are places are code that expects a certain type that is not immediately clear from the context, or when you declare fundamental types.
```

Comment:
Bad places are code that expects a certain type that is not immediately clear ...